components/PersonalFinance/Slider/IncomeExpense.js

```diff
@@ -13,34 +24,34 @@ export function IncomeExpense({ incomeMonthlyAvg, expenseMonthlyAvg }) {
           </div>
         </div>
         <div className="flex flex-col m-2 sm:m-6">
-          <div className="flex">
-            <div className="flex flex-col w-1/2 mr-2 sm:rounded-2xl sm:border-2 sm:border-[#F5F5F5] sm:bg-[#FEFEFE]">
-              <div className="mb-2 ml-2">
-                <span className="text-xs font-medium text-blue">This month</span>
-              </div>
-              <div className="flex items-end ml-2 sm:justify-center sm:ml-0">
-                <div>
-                  <img src="/slider/rectangle-1.svg" alt="Rectangle" />
-                  <span className="mr-1 font-bold text-[9px] sm:text-[11px] text-[#24CCA7]">Income</span>
+          <div className="flex items-center justify-center">
+            {incomeMonthlyAvg || expenseMonthlyAvg ? (
+              <div className="flex flex-col w-1/2 mr-2 sm:rounded-2xl sm:border-2 sm:border-[#F5F5F5] sm:bg-[#FEFEFE]">
+                <div className="ml-2">
+                  <span className="text-xs font-medium text-blue">This month</span>
+                </div>
+                {/* DESKTOP VIEW */}
+                <div className=" hidden sm:justify-center sm:ml-0 sm:block">
+                  <SingleBarChart data={data} width={110} height={120} left={40} right={40} />
                 </div>
-                <div>
-                  <img src="/slider/rectangle-2.svg" alt="Rectangle" />
-                  <span className="font-bold text-[9px] sm:text-[11px] text-[#4A56E2]">Expense</span>
+                {/* MOBILE VIEW */}
+                <div className="sm:hidden flex items-end ml-2">
+                  <SingleBarChart data={data} width={110} height={100} left={10} right={10} />
                 </div>
               </div>
-            </div>
+            ) : null}
```

Comment:
Actually if there is no data, whole chart and texts so this component should not be visible. So if you do that way there will be no need to use null @ardacoskun